traitlets/traitlets.py

```diff
@@ -974,6 +975,11 @@ def setup_instance(self, *args, **kwargs):
                 if isinstance(value, BaseDescriptor):
                     value.instance_init(self)
 
+def rollback_change(change):
+    if change.old is not Undefined:
+        change.owner.set_trait(change.name, change.old)
+    else:
+        change.owner._trait_values.pop(change.name)
```

Comment:
This will fail in the case that a default value was defined via `TraitType.__init__` since those are only assigned to the `HasTraits` instance during `instance_init`. I'm not sure this can be resolved given the current way default values are handled. However with #332, popping off of `_trait_values` would work.